### PR TITLE
Recreate SBOM

### PR DESCRIPTION
**Pull request by cdupuis**
# Description

This change reads the SBOM from the storage bucket in case it isn't provided in the input data.

## Related PRs

None

### CODE REVIEW

File: policy/policy_handler/subscription.go [hunk 1]
```diff
@@ -34,19 +34,47 @@ func getSubscriptionData(_ context.Context, req skill.RequestContext) (*goals.Ev
...
+func createSBOMFromManifest(ctx context.Context, subscriptionResult []map[edn.Keyword]edn.RawMessage) (*types.SBOM, error) {
+	imageEdn, ok := subscriptionResult[0][edn.Keyword("image")]
+
+	if !ok {
+		return nil, fmt.Errorf("image not found in subscription result")
+	}
+
+	image := util.Decode[goals.ImageSubscriptionQueryResult](imageEdn)
+
+	ref := image.ImageRepo.Repository
```
Comment (chrispatrick) [inline policy/policy_handler/subscription.go:59]: I think we should check if `ImageRepo` is nil. Subscriptions may not include it.

File: policy/policy_handler/subscription.go [hunk 2]
```diff
@@ -34,19 +34,47 @@ func getSubscriptionData(_ context.Context, req skill.RequestContext) (*goals.Ev
 		SubscriptionBasisT: req.Event.Context.Subscription.Metadata.AfterBasisT,
 	}
 
-	sbom, err := createSbomFromSubscriptionResult(evalMeta.SubscriptionResult, req)
+	sb, err := createSBOMFromManifest(ctx, evalMeta.SubscriptionResult)
+	if err == nil {
+		return evalMeta, req.Event.Context.Subscription.Configuration, sb, nil
```
Comment (chrispatrick) [inline policy/policy_handler/subscription.go:39]: Might be good to log when this succeeds or something?

**Review by chrispatrick — APPROVED** [review]

File: policy/storage/sbom.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	"net/http"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/atomist-skills/go-skill/policy/types"
+	"google.golang.org/api/googleapi"
+)
+
+const (
+	bucketName = "atm-prod-stored-manifests"
```
Comment (chrispatrick) [inline policy/storage/sbom.go:15]: Can we set this based on environment please? So we can test in staging?

File: go.mod [hunk 4]
```diff
@@ -114,6 +114,6 @@ require (
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20231120223509-83a465c0220f // indirect
 	google.golang.org/grpc v1.59.0 // indirect
 	google.golang.org/protobuf v1.31.0 // indirect
-	gopkg.in/yaml.v3 v3.0.1 // indirect
```
Comment (chrispatrick) [inline go.mod:117]: Weird gomod change?

**Review by chrispatrick — APPROVED** [review]
:shipit:

File: policy/storage/sbom.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cdupuis) [reply policy/storage/sbom.go]: They are being written in the registry-integrations for now. In the future coming from meta gross.

**Pull request merged**